README.md

```diff
@@ -1,14 +1,110 @@
-# Library of cryptographic functions for Dart
+# Cryptographic digest functions for Dart
 
-A set of cryptographic functions implemented in pure Dart.
+A set of cryptographic digest functions implemented in pure
+Dart. Cryptographic digest is also known as cryptographic hashing.
+
+The following digest/hashing algorithms are supported:
 
 * MD5
-* SHA1
-* SHA265
-* HMAC
-* Converting a list of bytes into a hex string
-* Converting a list of bytes into a base64-encoded string.
-* Converting a Base 64 encoded String into list of bytes.
+* SHA-1
+* SHA-256
+* HMAC (i.e. HMAC-MD5, HMAC-SHA1, HMAC-SHA256)
+
+## Usage
+
+### Digest on a single input
+
+Use the `sha256` property to obtain an instance of the `Sha256` class.
+Or use the `sha1` or `md5` properties to use those other digest
+algorithms.
+
+Invoke the `convert` method, passing the input data as a sequence of
+bytes (i.e. `List<int>`). It returns a `Digest` object, where the
+digest value can be obtained as bytes in a `List<int>` or encoded in
+hexadecimal as a `String`.
+
+```dart
+import 'package:crypto/crypto.dart';
+import 'dart:convert'; // for the UTF8.encode method
+
+  ...
```

Comment:
I try to make my example code snippets valid stand-alone Dart programs if possible, which means writing out a `main()` method.